m4opt/models/extinction/airmass.py

```diff
@@ -0,0 +1,334 @@
+from ..core import state
+from .core import BaseExtinction
+
+import numpy as np
+import astropy.units as u
+from astropy.coordinates import EarthLocation, AltAz
+from astropy.modeling import custom_model
+from astropy.modeling.models import Const1D, Tabular1D
+from astropy.table import QTable
+from astropy.utils.data import download_file
+
+path_to_extin = ('https://raw.githubusercontent.com/astropy/' +
+                 'specreduce-data/main/specreduce_data/' +
+                 'reference_data/extinction/')
+
+
+def read_extinction_table(placename):
+    """
+    extinction tables usually have
+    x in angstroms
+    y in mag per airmass
+    """
+    try:
+        tablename = AtmosphericExtinction.available_tables[placename]
+    except KeyError:
+        raise ValueError("Invalid placename {0}. Must be one of {1}".format(
+                        placename,
+                        AtmosphericExtinction.available_tables.keys()))
+
+    table = QTable.read(download_file(path_to_extin+tablename, cache=True),
+                        format='ascii', names=('wavelength', 'extinction'))
+
+    x = (table['wavelength']*u.Angstrom).to(
+        BaseExtinction.input_units['x'], equivalencies=u.spectral())
+    y = table['extinction']
+    return np.flipud(x), np.flipud(y)
+
+
+def simple_airmass(zen):
+    """
+    Simple airmass calculation assuming plane-parallel atmosphere.
+    Not recommended for zenith angles (z) > 75 degrees.
+
+    Input:
+    z : zenith angle (`astropy.units.Quantity`)
+    """
+    return 1./np.cos(zen)
+
+
+def KastenYoung_airmass(zen):
```

Comment:
Looking at the latest version I now see the point that if we are treating this as a primarily *internal* function, and the user interface is through the `Airmass` class, then there's really no need for using coordinate objects here.

I do however, suggest adding an `__all__` to clarify this - see my comment just after this one.

Reply:
@eteq: gotcha, thanks for the clarification! I have also accepted Leo's suggestion above to write a `BaseAirmass` and move the implementation of these internal functions into their respective `PlaneParallelAirmass` and `KastenYoungAirmass` classes. This should clarify that these functions are indeed internal to the class.

Having done that, however, I will put `__all__` into the `airmass.py` file, since there are several functions (other than the airmass calculations) that are internal only.